src/View/Helper/FormHelper.php

```diff
@@ -1167,9 +1170,14 @@ protected function _optionsOptions($fieldName, $options)
             return $options;
         }
 
+        if (substr($fieldName, -5) === '._ids') {
+            $fieldName = substr($fieldName, 0, -5);
+        } elseif (substr($fieldName, -3) === '_id') {
+            $fieldName = substr($fieldName, 0, -3);
+        }
         $fieldName = array_slice(explode('.', $fieldName), -1)[0];
         $varName = Inflector::variable(
-            Inflector::pluralize(preg_replace('/_id$/', '', $fieldName))
+            Inflector::pluralize($fieldName)
```

Comment:
Meh, looking at the tests, namely [`testInputOverridingMagicSelectType`](https://github.com/ndm2/cakephp/blob/3.0-improved-belongs-to-many-select-field-support/tests/TestCase/View/Helper/FormHelperTest.php#L2934) and [`testInputMagicSelectForTypeNumber`](https://github.com/ndm2/cakephp/blob/3.0-improved-belongs-to-many-select-field-support/tests/TestCase/View/Helper/FormHelperTest.php#L3033), it seems that the general pluralization to magically pick up any variable based on this convention is still wanted?